std/assembly/array.ts

```diff
@@ -476,6 +476,62 @@ export class Array<T> extends ArrayBufferView {
     return this;
   }
 
+  flat(): valueof<T>[] {
+    assert(isArray<T>());
```

Comment:
Changing this to `if (isArray<T>()) ERROR("something")` will result in a static compile error instead of an abort at runtime

Reply:
In this case `[1,2,3].flat()` will panic instead just return `[1,2,3]`. May be better?
```ts
if (!isArray<T>()) return this;
```

Reply:
We can't because the flat() method won't return the correct type.

Reply:
If I can cause this to become a compiler error that would make things great! I'll do that.

Reply:
Let me know if this change suffices. The error message could probably be made better. @dcodeIO

Reply:
The error message looks fine to me.